src/dmd/e2ir.d

```diff
@@ -6062,35 +6063,42 @@ Symbol *toStringSymbol(const(char)* str, size_t len, size_t sz)
             /* The stringTab pools common strings within an object file.
              * Win32 and Win64 use COMDATs to pool common strings across object files.
              */
-            import dmd.root.outbuffer : OutBuffer;
-            import dmd.dmangle;
-
-            scope StringExp se = new StringExp(Loc.initial, str[0 .. len], len, cast(ubyte)sz, 'c');
-
             /* VC++ uses a name mangling scheme, for example, "hello" is mangled to:
              * ??_C@_05CJBACGMB@hello?$AA@
              *        ^ length
              *         ^^^^^^^^ 8 byte checksum
              * But the checksum algorithm is unknown. Just invent our own.
              */
+
+            import dmd.root.outbuffer : OutBuffer;
             OutBuffer buf;
             buf.writestring("__");
-            mangleToBuffer(se, &buf);   // recycle how strings are mangled for templates
-
-            if (buf.length >= 32 + 2)
-            {   // Replace long string with hash of that string
-                import dmd.backend.md5;
-                MD5_CTX mdContext = void;
-                MD5Init(&mdContext);
-                MD5Update(&mdContext, cast(ubyte*)buf.peekChars(), cast(uint)buf.length);
-                MD5Final(&mdContext);
-                buf.setsize(2);
-                foreach (u; mdContext.digest)
+
+            void printHash()
+            {
+                // Replace long string with hash of that string
+                // Use the hash calculated in stringTab and a counter
+                __gshared uint count;
+                buf.writestring("ls");
+                buf.print(count++);
```

Comment:
Wouldn't this remove the mergeability of the string? Couldn't we use length and hash instead?

Reply:
Oh, you are right. Originally it was the length and I changed it to test better, but it lost the complete point of the hash.